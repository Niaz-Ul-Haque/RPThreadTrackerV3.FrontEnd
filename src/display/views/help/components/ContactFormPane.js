import React from 'react';
import {
	Row, Col, TabPane, Form, FormGroup, Input, Label, Button
} from 'reactstrap';

const ContactFormPane = () => (
	<TabPane tabId="contact">
		<Form action="" method="post" encType="multipart/form-data" className="form-horizontal">
			<p>Have a suggestion about the site? Encountered a bug? Want to just say hi or give me a hug? Please feel free to send me a message, or visit the tracker <a target="_blank" rel="noopener noreferrer" href="http://tblrthreadtracker.tumblr.com">support blog</a>.</p>
			<FormGroup row>
				<Col md="3">
					<Label htmlFor="email-input">Email</Label>
				</Col>
				<Col xs="12" md="9">
					<Input
						type="email"
						id="email-input"
						name="email-input"
						placeholder="Enter Email"
					/>
				</Col>
			</FormGroup>
			<FormGroup row>
				<Col md="3">
					<Label htmlFor="textarea-input">Message</Label>
				</Col>
				<Col xs="12" md="9">
					<Input
						type="textarea"
						name="textarea-input"
						id="textarea-input"
						rows="9"
						placeholder="Enter your message here..."
					/>
				</Col>
			</FormGroup>
		</Form>
		<Row>
			<Col className="text-right">
				<Button type="submit" color="primary">
					Submit
				</Button>
			</Col>
		</Row>
	</TabPane>
);

export default ContactFormPane;
