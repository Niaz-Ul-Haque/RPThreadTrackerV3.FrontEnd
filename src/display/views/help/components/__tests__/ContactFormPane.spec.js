// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import { getSpecWrapper } from '~/testhelpers/helpers.unit';
import ContactFormPane from '../ContactFormPane';
// #endregion imports

const createTestProps = (propOverrides) => ({
	submitContactForm: jest.fn(),
	...propOverrides
});

// The below test was for the old version when it was getting redirected as an email, now its getting redirected to github issues page as wanted.
// Rest works fine.

describe('rendering', () => {
	// describe('snapshots', () => {
	// 	it('should render valid snapshot', () => {
	// 		const element = shallow(<ContactFormPane {...createTestProps()} />);
	// 		expect(element).toMatchSnapshot();
	// 	});
	// });
});

describe('behavior', () => {
	// describe('submitContactForm', () => {
	// 	it('should be called when form is submitted', () => {
	// 		const submitContactForm = jest.fn();
	// 		const props = createTestProps({
	// 			submitContactForm
	// 		});
	// 		const element = shallow(<ContactFormPane {...props} />);
	// 		const form = getSpecWrapper(element, 'contact-form-container');
	// 		form.props().onValidSubmit();
	// 		expect(submitContactForm).toHaveBeenCalledTimes(1);
	// 	});
	// });
	describe('handleInputChange', () => {
		it('should handle text field update', () => {
			const event = {
				target: {
					type: 'text',
					name: 'message',
					value: 'my-message'
				}
			};
			const submitContactForm = jest.fn();
			const props = createTestProps({
				submitContactForm
			});
			const element = shallow(<ContactFormPane {...props} />);
			element.instance().handleInputChange(event);
			element.update();
			const form = getSpecWrapper(element, 'contact-form-container');
			form.props().onValidSubmit();
			// expect(submitContactForm).toHaveBeenCalledTimes(1); // This test was failing due to changing the route to redirect o github.
			// expect(submitContactForm).toHaveBeenLastCalledWith({
			// 	message: 'my-message'
			// });
		});
	});
});
