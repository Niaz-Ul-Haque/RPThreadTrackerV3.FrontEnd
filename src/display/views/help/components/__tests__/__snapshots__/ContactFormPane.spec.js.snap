// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<TabPane
  tabId="contact"
  tag="div"
>
  <Styled(Card)>
    <CardHeader
      tag="div"
    >
      <i
        className="fas fa-envelope"
      />
       Contact Me
    </CardHeader>
    <CardBody
      className="card-body"
      tag="div"
    >
      <AvForm
        data-spec="contact-form-container"
        method="get"
        model={Object {}}
        onInvalidSubmit={[Function]}
        onKeyDown={[Function]}
        onSubmit={[Function]}
        onValidSubmit={[Function]}
        tag={[Function]}
        validationEvent={
          Array [
            "onChange",
            "onInput",
          ]
        }
      >
        <p>
          Have a suggestion about the site? Encountered a bug? Want to just say hi or give me a hug? Please feel free to send me a message, or visit the tracker
           
          <a
            href="http://tblrthreadtracker.tumblr.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            support blog
          </a>
          .
        </p>
        <FormGroup
          row={true}
          tag="div"
        >
          <Col
            md="3"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Label
              htmlFor="textarea-input"
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              Message
            </Label>
          </Col>
          <Col
            md="9"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
            xs="12"
          >
            <Input
              id="message-textarea"
              name="Message"
              onChange={[Function]}
              placeholder="Enter your message here..."
              rows="9"
              type="textarea"
            />
          </Col>
        </FormGroup>
        <Row
          tag="div"
        >
          <Col
            className="text-right"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Button
              color="primary"
              tag="button"
              type="submit"
            >
              Submit
            </Button>
          </Col>
        </Row>
      </AvForm>
    </CardBody>
  </Styled(Card)>
</TabPane>
`;
